File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/persistence/candidate/CandidateRepository.java
package kyonggiyo.persistence.candidate;

import kyonggiyo.domain.candidate.Candidate;
import kyonggiyo.domain.candidate.Status;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Slice;

public interface CandidateRepository {

    Candidate getById(Long id);

    Slice<Candidate> findAllByStatus(Status status, Pageable pageable);

    Candidate save(Candidate candidate);

    void deleteById(Long id);

}

File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/candidate/repository/CandidateRepository.kt
package kyonggiyo.candidate.repository

import kyonggiyo.application.candidate.domain.entity.Candidate
import kyonggiyo.application.candidate.domain.vo.Status
import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Slice

interface CandidateRepository {

    fun getById(id: Long): Candidate

    fun findAllByStatus(status: Status, pageable: Pageable): Slice<Candidate>

    fun save(candidate: Candidate): Candidate

    fun deleteById(id: Long)

}
